src/main/java/com/eainde/assignment/EmployeeService.java

```diff
@@ -1,47 +1,121 @@
+            employeeList.add(new Employee(
+                    Long.parseLong(employeeDetails[0]),
+                    employeeDetails[1],
+                    Integer.parseInt(employeeDetails[2]),
+                    Double.parseDouble(employeeDetails[3]),
+                    employeeDetails[4],
+                    new Address(employeeDetails[5],employeeDetails[6],employeeDetails[7],employeeDetails[8])));
+        }
+
+        return employeeList;
     }
 
-    public List<Employee> sortedEmployees(){
-        // TODO Sort an Employee Object list first by salary and if salaries are equal, then by name
-        return null;
+    /*
+     * Orders list by salary ascending order and name
+     *
+     */
+
+    public List<Employee> sortedEmployees() throws IOException{
+
+        return this.getEmployees().stream()
+                .sorted(Comparator.comparing(Employee::getName))
+                .sorted(Comparator.comparing(Employee::getSalary).reversed()) // add for decending.reversed()
+                .collect(Collectors.toList());
+
     }
 
     public Employee getEmployeeById(long id){
-        // TODO write a code to get the employee by id
-        return null;
+
+
+        Employee emp = null;
+
+        try {
+
+            for(Employee employee: this.getEmployees()) {
+
+                if(employee.getId()==id) {
+                    emp = employee;
+                    break;
+                }
+
+            }
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+
+        return emp;
     }
 
-    public Employee getHighestSalaryEmployee(){
-        // TODO write a code to get the employee who has highest salary
-        return null;
+    /*
+     * Calls sorted list, takes last element of list, which is employee of higest salary
+     */
+
+    public Employee getHighestSalaryEmployee() throws IOException{
+
+        List<Employee> empList = this.sortedEmployees();
+        return empList.get(0);
     }
 
-    public int getAverageSalary(){
-        // TODO write a code to get the average salary of employees
-        return 0;
+    public double  getAverageSalary() throws IOException{
+
+        OptionalDouble average =
+                this.getEmployees().stream()
+                        .map(Employee::getSalary)
+                        .mapToDouble(a -> a)
+                        .average();
+
+        return Math.round(average.getAsDouble() * 100.0) / 100.0;
     }
 
-    public List<Employee> getEmployeesByCity(String city){
-        // TODO write a code to get the employees by city. If there are two employees for one city then they should be sorted by employee name
-        // Note: use HashMap
-        return null;
+
+    /*
+     * Add Employees from city to Hashmap and sort
+     */
+
+    public List<Employee> getEmployeesByCity(String city) throws IOException{
+
+        HashMap<String, Employee> hm = new HashMap<String, Employee>();
```

Comment:
Explicit type argument String, Employee can be replaced with <>

Reply:
Applying this rule reduces clutter and improves readability.

Reply:
I see